recipes/hdf5/all/conanfile.py

```diff
@@ -145,29 +147,60 @@ def package(self):
         os.remove(os.path.join(self.package_folder, "lib", "libhdf5.settings"))
 
     def package_info(self):
-        self.cpp_info.names["cmake_find_package"] = "HDF5"
-        self.cpp_info.names["cmake_find_package_multi"] = "HDF5"
-        self.cpp_info.libs = self._get_ordered_libs()
-        self.cpp_info.includedirs.append(os.path.join(self.package_folder, "include", "hdf5"))
+        self.cpp_info.names["cmake_find_package"] = "hdf5"
```

Comment:
There is an official module file built in CMake for hdf5: https://cmake.org/cmake/help/latest/module/FindHDF5.html
Hard to model.

Reply:
I know, would you just drop the one of hdf5?

Reply:
we should at least make the `find_package` work with the official name

Reply:
Sometimes better to drop module name in conan recipe, like in opengl recipe.

Reply:
> Sometimes better to drop module name in conan recipe, like in opengl recipe.

because it's not reasonable to get close, force them to use the conan flavor?

Reply:
cmake's `FindHDF5.cmake` uses hdf5's `HDF5-config.cmake` internally.

This fails because `FindHDF5.cmake` assumes that the generated targets are imported library targets. But cmake generates interface library targets.
The errors are:
```
-- Selected imported configuration: _imported_conf-NOTFOUND
CMake Error at /usr/share/cmake/Modules/FindHDF5.cmake:519 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "IMPORTED_IMPLIB__imported_conf-NOTFOUND" is not allowed.
Call Stack (most recent call first):
  CMakeLists.txt:7 (find_package)
```

Maybe, it would be nice if cmake would generate imported library targets if a target contains only one library, and a interface library if there are multiple.